### DISCUSSION TITLE
Dockerfile in project

### DISCUSSION THREAD

**Discussion by jfmario** [livebud/bud#227] [Q&A] [answered]
Here is a simple, 2-stage dockerfile.

```docker

FROM golang:1.18.4-buster AS build

# install node & npm
RUN apt-get update
RUN apt-get install -y curl g++
RUN curl sL https://deb.nodesource.com/setup_16.x | bash
RUN apt-get install -y nodejs

# install bud
RUN curl -sf https://raw.githubusercontent.com/livebud/bud/main/install.sh | VERSION=v0.2.1 sh

# build your app
COPY . /app
WORKDIR /app

RUN npm i
RUN bud build

FROM golang:1.18.4-buster

COPY --from=build /app /app

WORKDIR /app

ENV LOG_LEVEL=warn

EXPOSE 80

ENTRYPOINT ./bud/app --listen 0.0.0.0:80 --log $LOG_LEVEL

# docker run -it -e LOG_LEVEL=debug -p 8080:80 -<image-name>
```

Note: The downgrade to 0.2.1 is just a workaround to a regression that happens to exist as I'm working on this, and isn't necessarily otherwise.

Another note: I'm using ubuntu because I've had trouble running go v8 on alpine, if anyone can do this I'd be interested.

Anyways, is there interest in cleaning this up and dropping something like this into new projects on `bud create`, or on `bud add dockerfile` or something like that?

**Comment by matthewmueller** [comment, marked as answer]
Awesome, thanks for sharing! 

> Another note: I'm using ubuntu because I've had trouble running go v8 on alpine, if anyone can do this I'd be interested.

We'll be moving [to Goja soon](https://github.com/livebud/bud/discussions/188#discussioncomment-3213136) so all the V8 woes will no longer be a problem 😄 

>Anyways, is there interest in cleaning this up and dropping something like this into new projects on bud create, or on bud add dockerfile or something like that?

I think this should start out as a custom command plugin. Custom commands are something I'd like to return to after [v0.3](https://github.com/livebud/bud/discussions/188) is out. You should be able to add your own scaffolders.

Something like `command/add/dockerfile/command.go`:

```go
package dockerfile

type Command struct {
  // dependencies
}

type Flag struct {
  // flags
}

func (c *Command) Dockerfile(ctx context.Context, flag *Flag, arg1 int, arg2 string) error {
  // scaffold docker file
  return nil
}
```
